assets/prototype/domain/eventEditor/containers/datesListData/index.js

```diff
@@ -0,0 +1,81 @@
+import { createContext, useReducer, useEffect, useState } from '@wordpress/element';
+import gql from 'graphql-tag';
+import { useQuery } from '@apollo/react-hooks';
+
+const GET_DOGS = gql`
+	query getEventData($eventId: Number!) {
+		eventBy(eventId: $eventId) {
+			name
+		}
+	}
+`;
+
+const DEFAULT_STATE = {
+	datetimes: [],
+};
+
+const reducer = ( state, action ) => {
+	switch ( action.type ) {
+		case 'setDatetimes':
+			console.log( 'action.payload', action.payload );
+			return { datetimes: action.payload };
+		case 'add':
+			return { todoList: [ ...state.todoList, action.payload ] };
+	}
+};
+
+export const DatesListDataContext = createContext( {
+	state: DEFAULT_STATE,
+	dispatch: () => {},
+} );
```

Comment:
The purpose of prototype is to see how we are going to use Apollo and its API. We don't need to use context at all. The data is already in the cache. As we had discussed, we will pass the ID to each entity component and then get the data from cache.

Reply:
This is outdated, have been cleaned up here: https://github.com/eventespresso/event-espresso-core/pull/1876/commits/ed2145ef7002d1660cb2d3e3a893dce1e5c74b40

Reply:
The comment is about using `createContext` which is still there in the latest commit.

Reply:
Ok, I thought it's related to `state` and `dispatch`.